### PR TITLE
Add blog posts to search UI

### PR DESCRIPTION
**Pull request by marians**
This PR adds searching in the blog index, potentially returning hits for blog posts matching the search query.

Towards https://github.com/giantswarm/giantswarm/issues/15387

## Preview

image

### CODE REVIEW

**Comment by Oshratn** [comment]
LGTM - I like the fact that it is clear that the hit is a blog post and the publish date.

When there is a mixed bag - how will it appear? By date? By relevance? Broken up into sections (i.e. results from docs and results from blog)?

**Review by Oshratn — APPROVED** [review]

**Comment by marians** [comment]
> When there is a mixed bag - how will it appear? By date? By relevance? Broken up into sections (i.e. results from docs and results from blog)?

Right now, it's mixed (so that the changes to the search UI are minimal). The sorting is by relevance. Relevance is a bit more complex. Blog articles are by default less relevant than normal docs articles. But if a blog post fits the search entry much better than a docs article, it will come up higher.

File: src/assets/scripts/base.js [hunk 1]
```diff
@@ -171,9 +190,23 @@ function renderSerpEntry(index, hit) {
   } else {
     d.append($("<h4></h4>").html((index + 1) + ". " + hit._source.title));
   }
+
   if (typeof(hit._source.breadcrumb) !== "undefined" && hit._source.breadcrumb.length > 0) {
-    d.append($("<p class='sbreadcrumb'></p>").html("/" + hit._source.breadcrumb.join("/") + "/"));
+    var top = hit._source.breadcrumb[0].toLowerCase();
+    if (top === 'blog') {
+      var breadcrumb = 'Blog post'
+      if (typeof(hit._source.published) !== "undefined") {
```
Comment (ubergesundheit) [inline src/assets/scripts/base.js:198]: Did this work? IMO the syntax to check for undefined is like this:

```suggestion
      if (typeof hit._source.published !== "undefined") {
```

**Review by ubergesundheit — APPROVED** [review]
LGTM.

Regarding the `typeof` syntax it seems I was wrong. Both with and without parentheses is supported. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof

**Comment by marians** [comment]
> Regarding the typeof syntax it seems I was wrong.

😱 

Thanks for letting me know :)

**Pull request merged**